File: rs2-rent-sistem/rs2-rent-sistem.Model/Models/OrderItem.cs
namespace rs2_rent_sistem.Model.Models
{
    public class OrderItem
    {
        public int ID { get; set; }
        public int? Quantity { get; set; }
        public decimal? CostPerUse { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public double Price { get; set; }
        public virtual Equipment? Equipment { get; set; }
        public virtual ICollection<Review> Reviews { get; } = new List<Review>();

    }
}

File: rs2-rent-sistem/rs2-ui/rs2_rent_sistem/lib/models/order_item.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:rs2_rent_sistem/models/equipment_list_item.dart';

part 'order_item.freezed.dart';
part 'order_item.g.dart';

@Freezed()
class OrderItem with _$OrderItem {
  const factory OrderItem({
    required int id,
    required DateTime startDate,
    required DateTime endDate,
    @Default(0) int quantity,
    @Default(0) double costPerUse,
    required EquipmentListItem equipment,
  }) = _OrderItem;

  factory OrderItem.fromJson(Map<String, dynamic> json) => _$OrderItemFromJson(json);
}
